test/test.js

```diff
@@ -0,0 +1,132 @@
+const chai = require("chai"),
+      sinon = require("sinon"),
+      expect = chai.expect,
+      { assert } = require('chai'),
+      path = require('path'),
+      confFile = path.resolve(__dirname + '/../config.json'),
+      jsonfile = require('jsonfile'),
+      ContractClass = require('../modules/contract/contracts'),
+      AccountClass = require('../modules/accounts/accounts'),
+      contract = new ContractClass.Contracts(),
+      account = new AccountClass.Accounts();
+
+const CONTRACT_ADDRESS = 'KT1ABCDZ6krSvWKQXNXGRoQQwERm1eYtf123',
+      CONTRACT_LABEL = 'samplecontract',
+      BOOTSTRAPPED_ACCOUNT = 'tz1KqTpEZ7Yob7QbPE4Hy4Wo8fHG8LhKxZSx',
+      CONTRACT_INITIAL_STORAGE = "\"tezsure\"",
+      CONTRACT_INVOCATION_STORAGE = "\"tezster\"",
+      SET_LOCAL_PROVIDER = 'http://localhost:18731';
+
+describe('Test Network Provider', async () => {
+    it('set-provider', async () => {
+        //spy = sinon.spy(account, 'setProvider');
+        var mock = sinon.mock(account);
+        var expectation = mock.expects('setProviderAccounts');
+        expectation.exactly(1);
+        account.setProvider([SET_LOCAL_PROVIDER]);
+        mock.verify();
```

Comment:
I see all the tests are quite similar. Can you explaing what exactly are we trying to test here?

Reply:
That was the basic structure to use mostly functions from our code. Just creating the mock function from 3rd party used functions, will update code accordingly after successful mocking.